main.go

```diff
@@ -0,0 +1,543 @@
+		}
+		task.setReclaimTimer()
+		err = task.fetchTaskDefinition()
+		if err != nil {
+			log.Printf("WARN: Not able to fetch task definition for task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		err = task.validatePayload()
+		if err != nil {
+			log.Printf("WARN: Not able to validate task payload for task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		err = task.run()
+		if err != nil {
+			log.Printf("WARN: Not able to run task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		break
+	}
+	return taskFound
+}
+
+// Queries the given Azure Queue signed url pair (poll url/delete url) and
+// translates the Azure response into a Task object
+func (urlPair SignedURLPair) Poll() (*TaskRun, error) {
+	queueMessagesList := new(QueueMessagesList)
+	// To poll an Azure Queue the worker must do a `GET` request to the
+	// `signedPollUrl` from the object, representing the Azure queue. To
+	// receive multiple messages at once the parameter `&numofmessages=N`
+	// may be appended to `signedPollUrl`. The parameter `N` is the
+	// maximum number of messages desired, `N` can be up to 32.
+	// Since we can only process one task at a time, grab only one.
+	resp, err := http.Get(urlPair.SignedPollUrl + "&numofmessages=1")
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	// When executing a `GET` request to `signedPollUrl` from an Azure queue object,
+	// the request will return an XML document on the form:
+	//
+	// ```xml
+	// <QueueMessagesList>
+	//     <QueueMessage>
+	//       <MessageId>...</MessageId>
+	//       <InsertionTime>...</InsertionTime>
+	//       <ExpirationTime>...</ExpirationTime>
+	//       <PopReceipt>...</PopReceipt>
+	//       <TimeNextVisible>...</TimeNextVisible>
+	//       <DequeueCount>...</DequeueCount>
+	//       <MessageText>...</MessageText>
+	//     </QueueMessage>
+	//     ...
+	// </QueueMessagesList>
+	// ```
+	// We unmarshal the response into go objects, using the go xml decoder.
+	fullBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	reader := strings.NewReader(string(fullBody))
+	dec := xml.NewDecoder(reader)
+	err = dec.Decode(&queueMessagesList)
+	if err != nil {
+		return nil, err
+	}
+	if len(queueMessagesList.QueueMessages) == 0 {
+		return nil, errors.New("Zero tasks returned in Azure XML QueueMessagesList")
+	}
+	if size := len(queueMessagesList.QueueMessages); size > 1 {
+		return nil, fmt.Errorf("%v tasks returned in Azure XML QueueMessagesList, even though &numofmessages=1 was specified in poll url", size)
+	}
+
+	// at this point we know there is precisely one QueueMessage (== task)
+	qm := queueMessagesList.QueueMessages[0]
+
+	// To find the task referenced in a message the worker must base64
+	// decode and JSON parse the contents of the <MessageText> tag. This
+	// would return an object on the form: {taskId, runId}.
+	m, err := base64.StdEncoding.DecodeString(qm.MessageText)
+	if err != nil {
+		return nil, err
```

Comment:
If this fail, you should delete the azure queue message... otherwise another worker will pick it up and it'll keep being thrown around until the task deadline is reached where which is the timetolive for the message.